themes/tdr/login/error.ftl

```diff
@@ -0,0 +1,13 @@
+<#import "template.ftl" as layout>
+<@layout.registrationLayout displayMessage=false; section>
+    <#if section = "header">
+        ${msg("errorTitle")}
+    <#elseif section = "form">
+        <p class="govuk-body">${message.summary?no_esc}</p>
+        <#if client?? && client.baseUrl?has_content>
+            <p class="govuk-body"><a id="backToApplication" class="govuk-link" href="${properties.tdrHomeUrl}/dashboard">${kcSanitize(msg("backToApplication"))?no_esc}</a></p>
```

Comment:
What do the `kcSanitize` and `no_esc` do, and are they necessary? The `no_esc` one looks a bit concerning.

Reply:
`no_esc`: prevents auto-escaping - https://freemarker.apache.org/docs/ref_builtins_string.html#ref_builtin_no_esc
``: Allows sanitizing of html that uses Freemarker ?no_esc. This way, html can be allowed but it is still cleaned up for safety. Tags and attributes deemed unsafe will be stripped out - https://www.keycloak.org/docs-api/12.0/javadocs/org/keycloak/theme/KeycloakSanitizerMethod.html

Reply:
Thanks. Just trying to wrap my head around this - it sounds like it's safe to use it in this context because we're displaying our own text. I'd want to be careful about displaying user-entered text, but I don't think we're doing that on any of these pages.